django/contrib/auth/password_validation.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import unicode_literals
+
+import gzip
+import os
+import re
+from difflib import SequenceMatcher
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured, ValidationError
+from django.utils import lru_cache
+from django.utils.encoding import force_text
+from django.utils.html import format_html
+from django.utils.module_loading import import_string
+from django.utils.six import string_types
+from django.utils.translation import ugettext as _
+
+
+@lru_cache.lru_cache(maxsize=None)
+def get_default_password_validators():
+    return get_password_validators()
+
+
+def get_password_validators(validator_config=None):
+    validators = []
+    if validator_config is None:
+        validator_config = settings.AUTH_PASSWORD_VALIDATORS
+    for validator in validator_config:
+        try:
+            klass = import_string(validator['NAME'])
+        except ImportError:
+            msg = "Invalid NAME for a password validator: %s. Check your AUTH_PASSWORD_VALIDATORS setting."
+            raise ImproperlyConfigured(msg % validator['NAME'])
+        validators.append(klass(**validator.get('OPTIONS', {})))
+
+    return validators
+
+
+def validate_password(password, user=None, password_validators=None):
```

Comment:
That's a sensible design. However, the custom validator setup is expected to be quite uncommon: you only need this if you have multiple different sets of validators within the same Django project. So for most people, it's now as simple as calling `validate_password()` - which would get slightly more complicated in your proposed design. I also don't feel strongly, so I'll leave it as is.